### DISCUSSION TITLE
Custom Events break if filtered by e.g. device, browser or exit-page…

### DISCUSSION THREAD

**Discussion by DPangerl** [plausible/analytics#1002] [Self-Hosted Support] [answered]
On the plausible dashboard:
If I filter my results by device, browser or entry-page, my custom events won't show up correctly. 
The values for `unique` and `total` are not correct and the properties of my events don't show up. Instead, a loading indicator keeps spinning forever. 

The console throws an error

```console
Uncaught (in promise) ApiError: 
** (Clickhousex.Error) Code: 47, e.displayText() = DB::Exception: Missing columns: 'meta.key' while processing query: 
'SELECT session_id AS session_id FROM sessions AS s0 SAMPLE 10000000 WHERE ((domain = 'MYDOMAIN') AND ((timestamp >= '2021-05-06T22:00:00') AND (start < '2021-05-07T22:00:00')) AND (screen_size = 'Mobile')) AND (NOT has(meta.key, 'location'))', required columns: 'session_id' 'start' 'domain' 'timestamp' 'screen_size' 'meta.key', maybe you meant: ['session_id','start','domain','timestamp','screen_size'] (version 21.3.2.5 (official build))
```

If i remove the filter and reload the page, everything works fine again.
The only filter working is using a specific top-page. 

Has anyone of you an idea how to solve this?

Thank you!

Info:

Installed plausible according to: https://plausible.io/docs/self-hosting

**Comment by ukutaht** [comment, marked as answer]
Hey, thanks for reporting.

Can you reproduce the same issue on our live demo? https://plausible.io/plausible.io

**Reply by DPangerl** [reply]
Hey, thank you for the quick response.  👍 

Sorry, but I can't reproduce the error on the demo page, because none of the custom events there have properties. 
But you can take a look at my public dashboard. https://plausible.dieneueentwicklung.de/dev.dieneueentwicklung.de

The contact goal for example has some props listed. If you click on it you can either break them down by type or location. As long as there is no filter set, this works as intended. Then try to filter the report by device (or possibly anything but a top page) and it will break. The console throws the error mentioned above. 

Also, sometimes (I can't reproduce it) the values for the custom events are strangely summed up. Which means, the goals `total` without a filter is like `3`… But filtered (e.g. by device) it says "18"…

I don't know if this is important, but some of the tracking infos come from testing the website on localhost. 
If you need further infos just ask. 

Thanks!

**Reply by ukutaht** [reply]
Thanks for clarifying.

I tried to reproduce with one of our customers' public page that has custom props. Everything seems to work OK here: https://plausible.io/elementary.io?goal=Payment&screen=Desktop

This is pretty puzzling to me. I assume you're running `v.1.3.0`? I haven't done any bugfixes to the queries since then if I remember correctly.

I think I need to test this with a newer version of Clickhouse. We're running `20.8.7.15` in production. Could be caused by a version mismatch. If you have an easy way to downgrade to that version and test your setup that would be helpful.

**Reply by ukutaht** [reply]
Created an issue to track this: https://github.com/plausible/analytics/issues/1003

**Reply by DPangerl** [reply]
Hm… i couldn't find the exact version anywhere, but i simply cloned from `plausible/hosting` according to the docs. 

I've cloned the exact same repo on my local machine and downgraded clickhouse to your suggested version (`20.8.7.15`). All other settings are the same as the live deployment. And, yes… with this version, the props are working as expected. 

Do you think it is safe to downgrade clickhouse on the live system? Or would my existing data be lost?

**Reply by ukutaht** [reply]
I don't know if downgrading is an option. You can try cloning the data directory and attempt a downgrade if you're happy to mess about with it.

Alternatively you can wait until I fix the issue. No promises as to when it will happen but the next self-hosted release will come in autumn most likely. Sorry about that....
